Initial upload for event-detectors scripts

aasbra opened the pull request:
This script is intended for use by customers and the Operations Team, and includes validation and logging to help identify issues that need to be addressed.

This script only supports the following event detector types:
 *      LOW_LIMIT
 *      HIGH_LIMIT
 *      MULTISTATE_STATE
 *      BINARY_STATE
 *      NO_UPDATE
 *      UPDATE
If additional types are needed, this script can be updated.

aasbra committed:
Initial upload for create-event-detectors script

This script only supports the following event detector types:
 *      LOW_LIMIT
 *      HIGH_LIMIT
 *      MULTISTATE_STATE
 *      BINARY_STATE
 *      NO_UPDATE
 *      UPDATE
If additional types are needed, this script can be updated.

aasbra commented:
I'm converting this to a draft PR to give me a chance to make it compatible with Mango 5.1. I'll test and then merge those changes here.

aasbra committed:
Added feature detection before reloadDataPoint

This latest change tested on Mango 5.0.4 and 5.1.1

aasbra commented:
I added the feature detection Jared suggested and have tested this on 5.0.4 and 5.1.1.

jazdw commented on create-event-detectors.js:
```diff
@@ -0,0 +1,549 @@
+    }
+}
+
+const verbose = (logMessage) => {
```
I'm surprised calling this lambda works above this line. Functions are hoisted to the top of the file, a `const` is not.

jazdw replied:
Actually I guess this makes sense, it is in scope when it is called.

aasbra replied:
That was a carry-over from the original script. I thought it was a bit odd, but I left it in. I'll re-write this as a simple function to make it more readable.

jazdw reviewed (approved):
Thanks for sharing this, good documentation too.

aasbra commented on create-event-detectors.js:
```diff
@@ -0,0 +1,549 @@
+
+    //Validations
+    /*
+    //Why are we checking every value and failing if any of them are missing?
```
This comment block should have been removed. There were some validations that were incorrect from the original script that were wrong. I will remove this block.

aasbra committed:
Update create-event-detectors.js

Removed unnecessary comment block and replaced lambda functions to improve readability.

jazdw commented:
@aasbra go ahead and merge it, let me know if you can't. I didn't realize my reviews were being dismissed when you pushed.

aasbra committed:
Delete create_event_detectors/create_event_detectors.js

aasbra committed:
Delete create_event_detectors/event_detectors.csv

aasbra committed:
Delete create_event_detectors/Event-detector-handlers_to_link/README.md

aasbra committed:
Delete create_event_detectors/Event-detector-handlers_to_link/edit_event_detectors.js

aasbra committed:
Delete create_event_detectors/Event-detector-handlers_to_link/delete-event-detectors.js

aasbra committed:
Delete create_event_detectors/Event-detector-handlers_to_link/create_event_detectors.js

aasbra committed:
Uploading new create and edit scripts.

This version of the create script includes a change to set the resetDuration to 0 SECONDS for Mango 5.1
This version of the edit script is new and needs to be reviewed.

aasbra committed:
Uploading delete-event-detectors.js

This is a slightly modified version of the existing delete-event-detectors script. The old version was removing spaces from the data rows of the CSV, which was not necessary.

aasbra committed:
Delete create-event-detectors.js

Removing the file from this location to add it to a subdirectory for better organization.

aasbra commented:
@jazdw
It’s probably something I am missing, but I don’t see a way to rename the existing “create_event_detectors” directory in GitHub.  I was thinking of renaming it to something like “event-detector-management” or something similar. I've uploaded the latest versions of the create, edit, and delete scripts into that directory, but I think the directory should be renamed.

_edit-event-detectors.js_
 The edit script has not been reviewed yet.

_create-event-detectors.js_
This was modified slightly, adding a check for setResetDuration for Mango 5.1

_delete-event-detectors.js_
This was modified slightly with improvements from your feedback on the other scripts. Do we need to make any accommodations for different versions of Mango to restart data points after deleting the detector, or is that not necessary?

jazdw commented:
> @jazdw It’s probably something I am missing, but I don’t see a way to rename the existing “create_event_detectors” directory in GitHub. I was thinking of renaming it to something like “event-detector-management” or something similar.

Git has no concept of tracking changes to directories, only files. Basically you just need to move all of the files in that directory to a new directory.

> _delete-event-detectors.js_ This was modified slightly with improvements from your feedback on the other scripts. Do we need to make any accommodations for different versions of Mango to restart data points after deleting the detector, or is that not necessary?

Yes I think that you probably need to reload the points if you delete the detectors in old versions of Mango.

aasbra committed:
New folder and create-event-detectors.js

aasbra committed:
New folder and delete-event-detectors.js

aasbra committed:
New folder and edit-event-detectors.js

aasbra committed:
Delete create_event_detectors directory

Removing this directory as the name is no longer correct for the contents. A new directory has been added called event_detector_scripts that will replace it.
